sotodlib/tod_ops/flags.py

```diff
@@ -897,3 +916,176 @@ def get_stats(aman, signal, stat_names, split_subscans=False, mask=None, name="s
 
     info_aman = wrap_stats(aman, name, stats_arr, stat_names, merge)
     return info_aman
+
+def get_focalplane_flags(aman, merge=True, overwrite=True, invalid_flags_name='fp_flags'):
+    """
+    Generate flags for invalid detectors in the focal plane.
+        The tod.
+    merge : bool
+        If true, merges the generated flag into aman.
+    overwrite : bool
+        If true, write over flag. If false, don't.
+    invalid_flags_name : str
+        Name of flag to add to aman.flags if merge is True.
+
+    Returns
+    -------
+    msk_invalid_fp : RangesMatrix
+        RangesMatrix of invalid detectors in the focal plane.
+    """
+    # Available detectors in focalplane
+    xi_nan = np.isnan(aman.focal_plane.xi)
+    eta_nan = np.isnan(aman.focal_plane.eta)
+    x = Ranges(aman.samps.count)
+    msk_invalid_fp = RangesMatrix([
+        Ranges.ones_like(x) if Y else Ranges.zeros_like(x) 
+        for Y in flag_invalid_fp
+    ])
```

Comment:
`flag_invalid_fp` is not defined here yet

Reply:
(glitch from one of the merging steps?) -- corrected